gensim/sklearn_api/d2vmodel.py

```diff
@@ -159,7 +164,7 @@ def fit(self, X, y=None):
             The trained model.
 
         """
-        if isinstance([i for i in X[:1]][0], doc2vec.TaggedDocument):
+        if isinstance(self._get_first(X), doc2vec.TaggedDocument):
```

Comment:
Don't you want this instead?

```suggestion
        if isinstance(self._get_first(X[:1]), doc2vec.TaggedDocument):
```

Reply:
Done